Add textToSpeech functionality for reading the back of cards

veronika-vonschultz opened the pull request:
Add a text-to-speech plugin for reading the back of cards.

veronika-vonschultz committed:
Add textToSpeech button

veronika-vonschultz committed:
Moved out CSS and JS

veronika-vonschultz committed:
Bumped version

veronika-vonschultz committed:
Set to use UK female voice

veronika-vonschultz committed:
Updated copyright lines

veronika-vonschultz committed:
Resolved npx error

veronika-vonschultz committed:
Fix voice selection

Introduced a delay to speaking so that the voices have had time to load.

veronika-vonschultz committed:
Break too long line

veronika-vonschultz committed:
Bump version

veronika-vonschultz committed:
Merge remote-tracking branch 'upstream/main' into text-to-speech

veronika-vonschultz committed:
Bump minor version instead

veronika-vonschultz committed:
Add fallback if default voice is missing

veronika-vonschultz committed:
Update label on button

vonschultz commented on src/texttospeech.js:
```diff
@@ -0,0 +1,67 @@
+    }
+
+    if (filteredVoices.length === 0) {
+      console.log('No voice found for text-to-speech.')
```
This is probably too hidden. If the user has clicked "Play back side", the user would expect to hear something, or else see an error message. Maybe `window.alert()`?

vonschultz commented on src/html.php:
```diff
@@ -113,7 +113,9 @@ function print_card($card_id, $cardfront, $cardback, $backvisible = true)
-    echo "<p class=\"cardbacktitle\">" . $l["Back"] . "</p>\n";
+    echo "<p class=\"cardbacktitle\">" . $l["Back"] . 
+        "<button class=\"speakButton\" onclick=\"javascript:event.preventDefault(); 
+        readCardBack(" . $card_id . ")\">Play back side</button></p>\n";
```
Instead of "Play back side", make a new entry in `$l` and add translations in `html_en.php` and `html_sv.php`.

veronika-vonschultz committed:
Introduce window alert and move to translation files

veronika-vonschultz committed:
Add additional fallback voices for iOS

vonschultz committed:
Merge branch 'main' into text-to-speech

vonschultz committed:
Move speakButton styling to separate file

Let there be a texttospeech.css to accompany the texttospeech.js,
and move the speakButton styling there.

vonschultz committed:
Make texttospeech a plugin, add settings

Make the texttospeech functionality a plugin, so it can be enabled
or disabled on a lesson-by-lesson basis by the user. Add settings
to the lesson settings page, so you can choose your favorite language
and voice for each lesson.
